src/stelnet/board/commodity/table/ProfitCalculator.java

```diff
@@ -0,0 +1,45 @@
+package stelnet.board.commodity.table;
+
+import com.fs.starfarer.api.campaign.econ.CommodityOnMarketAPI;
+import com.fs.starfarer.api.campaign.econ.MarketAPI;
+import stelnet.board.commodity.price.DemandPrice;
+import stelnet.board.commodity.price.Price;
+import stelnet.board.commodity.price.SupplyPrice;
+import stelnet.util.TableCellHelper;
+
+public abstract class ProfitCalculator {
```

Comment:
There is nothing wrong with writing out a BigClassName.andItsBigDickFunctionEvenIfItTakes54CharactersInLength(). Making this abstract and extending it is architecturally not a good decision.

With having it abstract and extended, you will lock yourself out of re-using this code and end up extending the table class later on to get the desired functionality.

What I did initially for intel buttons (Loading the whole ProfitTableContent inside the intel button) was a stupid thing to do. It is equivalent to including a whole house inside my electronical key card, because that is the only way I can get an access to the function which open the door.

Always choose composition over inheritance and there is no reason to force composition if you do not benefit anything from the state. Play with inheritance when you are planning a design where you will benefit from liskovs substitution principle.

Reply:
I considered composition. I'd use it if I had unit tests. Inheritance was just faster. Anyway, that class is gone.